### PR TITLE
Added products_only argument (from java cli) to decode method.

### PR DESCRIPTION
**Pull request by vladvrabie**
This is useful in the use case of processing 1D code bars (no QR codes and the like).

### CODE REVIEW

**Comment by dlenski** [comment]
It looks like this is basically equivalent to setting a specific set of `possible_formats`, right?

https://github.com/zxing/zxing/blob/master/javase/src/main/java/com/google/zxing/client/j2se/DecoderConfig.java#L88-L93

**Comment by vladvrabie** [comment]
Exactly, but i didn't find that list :))
When I researched zxing, I managed to find it's list of command line parameters, so that guided me

File: zxing/__init__.py [hunk 1]
```diff
@@ -31,7 +31,8 @@ def __init__(self, classpath=None, java=None):
         else:
             self.classpath = os.path.join(os.path.dirname(__file__), 'java', '*')
 
-    def decode(self, filenames, try_harder=False, possible_formats=None, pure_barcode=False):
+
```
Comment (dlenski) [inline zxing/__init__.py:34]: Nitpick: can you take out the extra blank line added here?

**Comment by dlenski** [comment]
> This is useful in the use case of processing 1D code bars (no QR codes and the like).

This seems fine with me, though if your goal is to be able to read _any_ of the **1D** barcodes supported by ZXing decoder, then you'll want to add these additional possible formats.

https://github.com/zxing/zxing/blob/master/javase/src/main/java/com/google/zxing/client/j2se/DecoderConfig.java#L88-L100

File: zxing/__init__.py (hunk 1 above)
Comment (vladvrabie) [reply zxing/__init__.py]: Yes

**Comment by vladvrabie** [comment]
> > This is useful in the use case of processing 1D code bars (no QR codes and the like).
> 
> This seems fine with me, though if your goal is to be able to read _any_ of the **1D** barcodes supported by ZXing decoder, then you'll want to add these additional possible formats.
> 
> https://github.com/zxing/zxing/blob/master/javase/src/main/java/com/google/zxing/client/j2se/DecoderConfig.java#L88-L100

Good to know, thank you. I will research those as well to see if they fit my use case

**Review by dlenski — APPROVED** [review]

**Pull request merged**